app/src/main/java/com/example/drawingApp/utils/DialogUtility.kt

```diff
@@ -124,26 +139,37 @@ object DialogUtility {
             )
         }
         tools.forEach { tool ->
-            view.findViewById<View>(tool.btn).setOnClickListener {
+            (tool.btn).setOnClickListener {
                 onSheetButtonClicked(tool)
             }
         }
-
         if (!state.isInitialized) {
-            showTabContent(view, tools, state.selectedTool)
+            when (state.selectedTool) {
+                ToolEnum.Gradient -> tools.find { it.tool == ToolEnum.Gradient }
```

Comment:
Why when block this? Each block is just using `.find` on the value. Why not just `tools.find { it.tool == state.selectedTool }`?

Reply:
when block was on my mind from what we talked about the day before when I was working on this, so instead of doing a good solution I tried to find a solution with a when block

Reply:
but yea the when block is super redundant will fix